scripts/migration/migrate_primary_institutions.py

```diff
@@ -0,0 +1,49 @@
+import sys
+import logging
+
+from modularodm import Q
+
+from website.app import init_app
+from website.models import Node, NodeLog
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.WARN)
+
+PRIMARY_INSTITUTION_CHANGED = 'primary_institution_changed'
+PRIMARY_INSTITUTION_REMOVED = 'primary_institution_removed'
+
+def migrate_logs(dry_run):
+    added_logs = NodeLog.find(Q('action', 'eq', PRIMARY_INSTITUTION_CHANGED))
+    for log in added_logs:
+        logger.warn('Log with id <{}> being updated for affiliation added'.format(log._id))
+        if not dry_run:
+            log.action = NodeLog.AFFILIATED_INSTITUTION_ADDED
+            log.save()
+    removed_logs = NodeLog.find(Q('action', 'eq', PRIMARY_INSTITUTION_REMOVED))
+    for log in removed_logs:
+        logger.warn('Log with id <{}> being updated for affiliation removed'.format(log._id))
+        if not dry_run:
+            log.action = NodeLog.AFFILIATED_INSTITUTION_REMOVED
+            log.save()
+
+def migrate_nodes(dry_run):
+    nodes = Node.find(Q('primary_institution', 'ne', None))
+    for node in nodes:
+        logger.warn('Node with id <{}> and title <{}> being updated'.format(node._id, node.title))
+        if not dry_run:
+            inst = node.primary_institution
+            if inst not in node.affiliated_institutions:
+                node.affiliated_institutions.append(inst)
+            node.primary_institution = None
+            node.save()
+
+def main(dry_run=True):
```

Comment:
You need to add a file logger if it's not a `dry_run`. Also, the migration itself should be run within a `TokuTransaction`. This prevents changes from persisting if an exception is thrown during the migration, and during a `dry_run` allows you to actually run the migration to ensure it works.

```python
import logging
import sys

from framework.transactions.context import TokuTransaction
from website.app import init_app
from scripts import utils as script_utils

logger = logging.getLogger(__name__)

def migrate(dry_run=True):
    # iterate over targets
    # perform actual migration
    # log things

    if dry_run:
        raise RuntimeError('Dry run, transaction rolled back.')

def main():
    dry_run = False
    if '--dry' in sys.argv:
        dry_run = True
    if not dry_run:
        script_utils.add_file_logger(logger, __file__)
    init_app(set_backends=True, routes=False)
    with TokuTransaction():
        migrate(dry_run=dry_run)

if __name__ == "__main__":
    main()
```